standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```
@@ -598,7 +632,9 @@ void truncateTable(String catName, String dbName, String tableName, List<String>
    *                isPurge flag when set to true files which needs to be recycled are not moved to Trash
    * @return Response which is currently void
    */
-  CmRecycleResponse recycleDirToCmPath(CmRecycleRequest request) throws MetaException, TException;
+  default CmRecycleResponse recycleDirToCmPath(CmRecycleRequest request) throws MetaException, TException{
+    return new CmRecycleResponse();
```

should throw UnsupportedOperationException